quickwit-janitor/src/actors/delete_task_service.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::{HashMap, HashSet};
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use quickwit_actors::{Actor, ActorContext, ActorExitStatus, ActorHandle, Handler, HEARTBEAT};
+use quickwit_metastore::{IndexMetadata, Metastore};
+use quickwit_search::SearchClientPool;
+use quickwit_storage::StorageUriResolver;
+use serde::{Deserialize, Serialize};
+use tracing::{error, info};
+
+use super::delete_task_pipeline::DeleteTaskPipeline;
+
+pub const DELETE_SERVICE_TASK_DIR_NAME: &str = "delete_task_service";
+
+#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+pub struct DeleteTaskServiceState {
+    pub num_running_pipelines: usize,
+}
+
+pub struct DeleteTaskService {
+    metastore: Arc<dyn Metastore>,
+    search_client_pool: SearchClientPool,
+    storage_resolver: StorageUriResolver,
+    data_dir_path: PathBuf,
+    pipeline_handles_by_index_id: HashMap<String, ActorHandle<DeleteTaskPipeline>>,
+    state: DeleteTaskServiceState,
```

Comment:
```suggestion
```

Please remove the state from `DeleteTaskService` and compute it on the fly on calls to
`observalbe_state`.

Right now the invariant `num_running_pipelines == self.pipeline_handles_by_index_Id` is needlessly difficult to proofread.

Reply:
yes indeed, I should have done it :/